ui/raidboss/raidemulator.js

```diff
@@ -29,398 +29,382 @@ import defaultOptions from './raidboss_options';
+      let target = e.currentTarget;
+      while (!target.classList.contains('modal') && target !== document.body)
+        target = target.parentElement;
 
-      // Handle closing all modals if the user clicks outside the modal
-      document.querySelectorAll('.modal').forEach((n) => {
-        n.addEventListener('click', (e) => {
-          // Only close the modal if the user actually clicked outside it, not child clicks
-          if (e.target === n)
-            hideModal();
-        });
-      });
+      if (target !== document.body)
+        hideModal('.' + [...target.classList].join('.'));
+    });
+  });
 
-      // Ask the user if they're really sure they want to clear the DB
-      document.querySelector('.clearDBButton').addEventListener('click', (e) => {
-        showModal('.deleteDBModal');
-      });
+  // Handle closing all modals if the user clicks outside the modal
+  document.querySelectorAll('.modal').forEach((n) => {
+    n.addEventListener('click', (e) => {
+      // Only close the modal if the user actually clicked outside it, not child clicks
+      if (e.target === n)
+        hideModal();
+    });
+  });
 
-      // Handle user saying they're really sure they want to clear the DB by wiping it then
-      // refreshing the encounter tab
-      document.querySelector('.deleteDBModal .btn-primary').addEventListener('click', (e) => {
-        persistor.clearDB().then(() => {
-          encounterTab.refresh();
-          hideModal('.deleteDBModal');
-        });
-      });
+  // Ask the user if they're really sure they want to clear the DB
+  document.querySelector('.clearDBButton').addEventListener('click', (e) => {
+    showModal('.deleteDBModal');
+  });
 
-      // Make the emulator state available for debugging
-      window.raidEmulator = {
-        emulator: emulator,
-        progressBar: progressBar,
-        timelineController: timelineController,
-        popupText: popupText,
-        persistor: persistor,
-        encounterTab: encounterTab,
-        emulatedPartyInfo: emulatedPartyInfo,
-        emulatedMap: emulatedMap,
-        emulatedWebSocket: emulatedWebSocket,
-        timelineUI: timelineUI,
-      };
+  // Handle user saying they're really sure they want to clear the DB by wiping it then
+  // refreshing the encounter tab
+  document.querySelector('.deleteDBModal .btn-primary').addEventListener('click', (e) => {
+    persistor.clearDB().then(() => {
+      encounterTab.refresh();
+      hideModal('.deleteDBModal');
     });
   });
-})();
+
+  // Make the emulator state available for debugging
+  window.raidEmulator = {
+    emulator: emulator,
+    progressBar: progressBar,
+    timelineController: timelineController,
+    popupText: popupText,
+    persistor: persistor,
+    encounterTab: encounterTab,
+    emulatedPartyInfo: emulatedPartyInfo,
+    emulatedMap: emulatedMap,
+    emulatedWebSocket: emulatedWebSocket,
+    timelineUI: timelineUI,
+  };
+});
 
 function showModal(selector) {
```

Comment:
Might make sense to move the modal functions to the top of the file now from a flow standpoint?

Reply:
Thanks, done!